enhancements/cluster-logging/loki-storage.md

```diff
@@ -0,0 +1,605 @@
+- No individual control of any cluster API to tune any operations.
+- Provides consistent schema-less model compatible with the rest of our observability tools, i.e. Prometheus, Thanos.
+
+In summary offering an alternative low-retention store based on Loki in cluster logging provides many new benefits in lifting the whole stack to a more cloud-native first experience. This has the potential to lessen the operational and the support burden by a lot.
+
+## Motivation
+
+Currently cluster logging is restricted to providing support for Elasticsearch as a log store. This proposal outlines the API in form of a custom resource definition (CRD) for an alternative log store based on Loki and managed by the [loki-operator](https://github.com/viaq/loki-operator).
+
+In addition it describes how to integrate the custom resource definition into the ClusterLogging custom resource that serves as the single entry point for installing the whole cluster logging stack.
+
+### Goals
+
+The specific goals of this proposals are:
+
+* Outline the API to provide a declarative CRD for Loki deployments managed by the [loki-operator](https://github.com/viaq/loki-operator).
+
+* Propose API support into the ClusterLogging CRD for single entry point deployments.
+
+We will be successful when:
+
+* Users are able to declare a Loki instance via the CRD and the [loki-operator](https://github.com/viaq/loki-operator) rolls out the Loki deployment.
+
+* Users are able to declare Loki as alternative store in the Cluster Logging custom resource.
+
+* Users are able to operate Loki in Cluster Logging without the need to tune every single bit via the proposed CRD.
+
+### Non-Goals
+
+* We will not provide a general purpose API to install Loki at the field. The API is dedicated to cluster logging only.
+
+* We will not expose or implement all possible Loki configuration knobs in the API besides the most crucial to operate in cluster logging.
+
+* We will not expose Loki as a general log store for other ingestion sources than the ones supported and managed by cluster logging.
+
+* We will not offer support for custom labeling of logs other than the defaults provided by the cluster logging collector.
+
+## Proposal
+
+To offer a seamless cluster logging experience with Loki as an alternative store, we propose to limit the API support to the following key features:
+
+* Deploy Loki **only** in microservices mode, i.e. each component is deployed and managed as an individual deployment/statefulset. This enables individual capacity planning and resource utilization per path ingestion/querying.
+
+* Multi-tenancy is enabled by default and requires the collector to apply the required [tenant identification](https://grafana.com/docs/loki/latest/operations/multi-tenancy/) on ingestion time. No user can disable this setting. In turn queries require the same identification information to be successful. Furthermore there is no support for querying across tenants.
```

Comment:
Asking out of my own ignorance here, is this a limit by Loki itself or are we imposing that? Also, is there some scenario in which a customer for some reason has would have multiple tenants assigned under it and thus might need such a feature?

Reply:
The current collector proposal allows the user to set a fixed tenant ID, use the namespace or use a label value.  Do you think we should default to tenant==namespace, or enforce it as the only option?

Reply:
@alanconway The default case should be `tenant == namespace` to keep the multi-tenancy model aligned with what we support with ES right now.